edx_exams/apps/core/management/commands/bulk_add_course_staff.py

```diff
@@ -0,0 +1,96 @@
+"""Management command to populate the CourseStaffRole model and related User objects, from LMS, using CSV"""
+import logging
+import time
+
+import unicodecsv
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from edx_exams.apps.core.models import CourseStaffRole, User
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Management command to add Course Staff (and User) in batches from CSV
+    """
+    help = """
+    Add Course Staff in bulk from CSV.
+    Expects that the data will be provided in a csv file format with the first row
+    being the header and columns being: username, email, role, course_id.
+    Example:
+        $ ... bulk_add_course_staff --csv_path=foo.csv
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-p', '--csv_path',
+            metavar='csv_path',
+            dest='csv_path',
+            required=False,
+            help='Path to CSV file.')
+        parser.add_argument(
+            '--batch_size',
+            type=int,
+            default=10000,
+            dest='batch_size',
+            help='Batch size')
+        parser.add_argument(
+            '--batch_delay',
+            type=float,
+            default=1.0,
+            dest='batch_delay',
+            help='Time delay in each iteration')
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        """
+        The main logic and entry point of the management command
+        """
+        csv_path = options['csv_path']
+        batch_size = options['batch_size']
+        batch_delay = options['batch_delay']
+
+        if csv_path:
+            with open(csv_path, 'rb') as csv_file:
+                self.add_course_staff_from_csv(csv_file, batch_size, batch_delay)
+
+        logger.info('Bulk add course staff complete!')
+
+    def add_course_staff_from_csv(self, csv_file, batch_size, batch_delay):
+        """
+        Add the given set of course staff provided in csv
+        """
+        reader = list(unicodecsv.DictReader(csv_file))
+        # todo: filter by users that haven't been created
```

Comment:
Will remove this comment, but just wanted to call out how I am filtering out users that have already been created. Lemme know if there is anything I can do differently. It did seem that using exists() was the most efficient way to check this.

Reply:
Oh cool! Yes, Michael did share this code as an example here: [manual_verifications](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/verify_student/management/commands/manual_verifications.py#L100) management command. But I didn't understand that using __in will help it be more efficient! So I'll update to make more similar to this example.